src/clj_kondo/impl/core.clj

```diff
@@ -361,6 +361,47 @@
 (defn classpath? [f]
   (str/includes? f path-separator))
 
+(defn ^:private entries-from-jar [^java.io.File jar-file]
+  (with-open [jar (JarFile. jar-file)]
+    (->> (.entries jar)
+         enumeration-seq
+         (filter (fn [^JarFile$JarFileEntry x]
+                   (and (not (.isDirectory x))
+                        (source-file? (.getName x))))))))
+
+(defn ^:private files-count [paths]
+  (->> paths
+       (map (fn [path]
+              (let [path (str path)
+                    file (io/file path)
+                    canonical (when (.exists file)
+                                (.getCanonicalPath file))]
+                (cond
+                  (and canonical
+                       (str/ends-with? canonical ".jar"))
+                  (count (entries-from-jar file))
+
+                  (and canonical
```

Comment:
The canonical stuff doesn't really matter for counting files I think?

Reply:
yes, this is just the path, maybe we could rename from canonical to other name?

Reply:
ok